app/src/main/java/com/example/dogadoption/repository/DogRepository.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.dogadoption.repository
+
+import android.app.Application
+import com.example.dogadoption.retrofit.DogApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class DogRepository @Inject constructor(
+    private val dogApi: DogApi,
+    private val appContext: Application
```

Comment:
What's up with this `Application` here?

In frontend development (mobile/web) you always have a few layers: UI, Business and Data or whatever it's called. The Application belongs to the UI layer and only this layer should know about it.